### PR TITLE
Enhance: supporting float in commit/reveal subtensor

### PR DESCRIPTION
**Pull request by ibraheem-abe**
- We are now normalizing weights in subtensor
- Allows the community to call commit/reveal with floats

### CODE REVIEW

**Review by thealligatorking — APPROVED** [review]

**Review by opendansor — APPROVED** [review]

**Review by thewhaleking — APPROVED** [review]

File: bittensor/subtensor.py [hunk 1]
```diff
@@ -912,8 +912,8 @@ def commit_weights(
         wallet: "bittensor.wallet",
         netuid: int,
         salt: List[int],
-        uids: Union[NDArray[np.int64], list],
-        weights: Union[NDArray[np.float32], list],
+        uids: Union[NDArray[np.int64]],
+        weights: Union[NDArray[np.float32]],
```
Comment (thewhaleking) [inline bittensor/subtensor.py]: Why the use of `Union` here? The typing here implies `uids` will only be an NDArray of int64, and weight only an NDArray of float32.

File: bittensor/subtensor.py [hunk 2]
```diff
@@ -1053,8 +1053,8 @@ def reveal_weights(
         self,
         wallet: "bittensor.wallet",
         netuid: int,
-        uids: Union[NDArray[np.int64], list],
-        weights: Union[NDArray[np.float32], list],
+        uids: Union[NDArray[np.int64]],
+        weights: Union[NDArray[np.float32]],
```
Comment (thewhaleking) [inline bittensor/subtensor.py]: Same thing here.

File: bittensor/subtensor.py (hunk 1 above)
Comment (ibraheem-abe) [reply bittensor/subtensor.py]: Great catch, thanks

**Comment by ibraheem-abe** [comment]
Converting this to draft as per the team's discussion to roll this out with the other commit/reveal improvements.

**Comment by ibraheem-abe** [comment]
Closing this we are now working on a revamped commit-reveal command

**Pull request closed without merging**